File: 1936.c
#include <stdio.h>

int main() {
    int v[] = {1, 2, 6, 24, 120, 720, 5040, 40320};
    int N, i, count = 0;

    scanf("%d", &N);

    for (i = 7; i >= 0; i--) {
        while (v[i] <= N) {
            N -= v[i];
            count++;
        }
        if (N == 0)
            break;
    }

    printf("%d\n", count);
    return 0;
}

File: 1936.py
v = [1, 2, 6, 24, 120, 720, 5040, 40320]
N = int(input())
i = count = 0

for i in range(7, -1, -1):
    while v[i]  <= N:
        N -= v[i]
        count += 1
    
    if N == 0:
        break

print(count)